conans/model/conf.py

```diff
@@ -493,13 +490,24 @@ def set_relative_base_folder(self, folder):
 
     @staticmethod
     def _check_conf_name(conf):
-        if USER_CONF_PATTERN.match(conf) is None and conf not in BUILT_IN_CONFS:
+        if CORE_CONF_PATTERN.match(conf) is not None or TOOLS_CONF_PATTERN.match(conf) is not None:
+            if conf in BUILT_IN_CONFS:
+                return  # It's a built-in conf
+            else:
+                raise ConanException(f"[conf] '{conf}' does not exist in configuration list. "
+                                     f" Run 'conan config list' to see all the available confs.")
```

Comment:
Why this necessary? The built-in confs are already fully validated, so this seems redundant?

Reply:
At this point the only validation done is the one a in the outter if - unknown confs can get here

Reply:
Ok, maybe just 
```python
if USER_CONF_PATTERN.match(conf):
     ...
elif conf not in BUILTIN_CONFS:
     raise
```
Would be enough?